### PR TITLE
add input type file

### LINKED ISSUE #87: Create functionality: Upload profile picture
imagen

### PR DESCRIPTION
**Pull request by AdrianFagilde**
create a component and function to update the image avatar

### CODE REVIEW

**Comment by AdrianFagilde** [comment]
@victor-heliomar  @Eghinner  @angelik0828 , please check this pull, i think it's ready

File: src/actions/UserActions.ts [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+import {
+  USER_LOGIN,
+  USER_BY_ID,
+  UPDATE_USER_PASSWORD,
+  UPDATE_USER_AVATAR,
+} from '../services';
 import { useMutation } from '@vue/apollo-composable';
 import { useQuery, Users, User, getUserReturnType } from '../utils';
 import { logger } from '../utils/logger';
+import axios from 'axios';
```
Comment (heliomar-pena) [inline src/actions/UserActions.ts]: Please remove axios from here since it's not being used

File: src/pages/User/components/UserHeader/UserHeader.vue [hunk 2]
```diff
@@ -57,15 +63,31 @@ import userDefaultImg from 'src/assets/images/user-default.svg';
...
+const showAvatar: Ref<string | undefined> = ref('');
 const props = defineProps<UserHeaderProps>();
+const emits = defineEmits<UserHeaderEmits>();
+showAvatar.value = props.avatar ? props.avatar : userDefaultImg;
+const uploadImage = async (event: Event) => {
+  const file = (event.target as HTMLInputElement).files?.item(0);
+  const uploadedImage = await handleAvatarUpload(file);
+  emits('saveAvatar', uploadedImage);
+  showAvatar.value = uploadedImage;
+};
```
Comment (heliomar-pena) [inline src/pages/User/components/UserHeader/UserHeader.vue]: Why is this event two times? Here and in `src/pages/User/components/UserHeader/utils/handleAvatarUpload.ts` file?

Both are needed? Can one of them to be removed?

File: src/pages/User/components/UserHeader/utils/handleAvatarUpload.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    const url = 'https://www.did-admin.tk/api/files?prefix=user-profile';
+    const formData = new FormData();
+    formData.append('file', file);
+    const uploadImage = await axios
+      .post(url, formData)
+      .then((res) => res.data)
+      .catch((err) => console.error(err));
+
+    const user = LocalStorage.getItem('user') as User;
+    LocalStorage.clear();
```
Comment (heliomar-pena) [inline src/pages/User/components/UserHeader/utils/handleAvatarUpload.ts]: Maybe instead of clear you could just remove 'user'

File: src/pages/User/utils/handleUserAvatar.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    isLoadingAvatar.value = true;
+
+    return new Promise<void>((resolve, reject) => {
+      uploadUserAvatar(id, avatar)
+        .then(() => {
+          newAvatar.value = '';
+
+          Notify.create({
+            type: 'positive',
+            message: 'La contraseña se actualizó correctamente.',
```
Comment (heliomar-pena) [inline src/pages/User/utils/handleUserAvatar.ts]: Please change the message

File: src/pages/User/utils/handleUserAvatar.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        });
+    });
+  };
+
+  return {
+    newAvatar,
+    isLoadingAvatar,
+    onUpdateUserAvatar,
+  };
+};
```
Comment (heliomar-pena) [inline src/pages/User/utils/handleUserAvatar.ts]: What is the difference between this file and `src/pages/User/components/UserHeader/utils/handleAvatarUpload.ts`?

File: src/pages/User/components/UserHeader/utils/handleAvatarUpload.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import axios from 'axios';
+import { User } from 'src/utils/apollo.types';
+import { LocalStorage } from 'quasar';
+
+export const handleAvatarUpload = async (file: File | null | undefined) => {
+  if (!file) return null;
+
+  if (file.type.includes('image')) {
+    const url = 'https://www.did-admin.tk/api/files?prefix=user-profile';
```
Comment (heliomar-pena) [inline src/pages/User/components/UserHeader/utils/handleAvatarUpload.ts]: By the moments please add this in the object "env" of the quasar.config.js file, which will be changed in another PR for a .env file

imagen

So if you add it some like this:

```js
env: {
   GRAPHQL_URL: ......,
   NEXT_URL: "https://www.did-admin.tk/api"
}
```
You could use it here some like this:

```js
const url = process.env.NEXT_URL + "/files?prefix=user-profile"
```

**Review by heliomar-pena — CHANGES_REQUESTED** [review]
Good work! It works properly
imagen

I just have some comments

File: src/pages/User/components/UserHeader/UserHeader.vue (hunk 2 above)
Comment (AdrianFagilde) [reply src/pages/User/components/UserHeader/UserHeader.vue]: This function seems to me necessary to be able to update the api and localstorage to visualize the change.

but you think we can do it in one, you could try to optimize the code.

**Comment by AdrianFagilde** [comment]
@victor-heliomar , please take a look at the changes and let me know if this is ok.

File: src/pages/User/components/UserHeader/utils/handleAvatarUpload.ts [hunk 7]
```diff
@@ -15,10 +19,49 @@ export const handleAvatarUpload = async (file: File | null | undefined) => {
       .catch((err) => console.error(err));
 
     const user = LocalStorage.getItem('user') as User;
-    LocalStorage.clear();
+    LocalStorage.remove('user');
     user.avatar_url = uploadImage.data.fileUrl;
     LocalStorage.set('user', user);
```
Comment (heliomar-pena) [inline src/pages/User/components/UserHeader/utils/handleAvatarUpload.ts]: I know that I suggested to remove 'user' hehe, but now that I think it, what about just setting the 'user' instead of remove it and add it again?

I think that with the LocalStorage.set('user', user) is enough, please check

File: src/pages/User/components/UserHeader/utils/handleAvatarUpload.ts [hunk 8]
```diff
@@ -15,10 +19,49 @@ export const handleAvatarUpload = async (file: File | null | undefined) => {
       .catch((err) => console.error(err));
 
     const user = LocalStorage.getItem('user') as User;
```
Comment (heliomar-pena) [inline src/pages/User/components/UserHeader/utils/handleAvatarUpload.ts]: Please use the util "getUser" 

https://github.com/Azordev/did-app/blob/14385ef549ac8efc0d8a74f76b5b181b740135c6/src/utils/auth.ts#L5-L20

File: src/pages/User/components/UserHeader/utils/handleAvatarUpload.ts [hunk 9]
```diff
@@ -15,10 +19,49 @@ export const handleAvatarUpload = async (file: File | null | undefined) => {
...
+    isLoadingAvatar.value = true;
+
+    return new Promise<void>((resolve, reject) => {
+      uploadUserAvatar(id, avatar)
+        .then(() => {
+          newAvatar.value = '';
+
+          Notify.create({
+            type: 'positive',
+            message: 'La Imagen se actualizó correctamente.',
```
Comment (heliomar-pena) [inline src/pages/User/components/UserHeader/utils/handleAvatarUpload.ts]:
```suggestion
            message: 'La imagen se actualizó correctamente.',
```

**Review by heliomar-pena — APPROVED** [review]
Looks good!

@angelik0828 please check if works

File: src/pages/User/components/UserHeader/UserHeader.vue (hunk 2 above)
Comment (heliomar-pena) [reply src/pages/User/components/UserHeader/UserHeader.vue]: It's ok so, thanks for confirm!

**Review by heliomar-pena — APPROVED** [review]
good work! @AdrianFagilde 

@angelik0828 please check

**Comment by Eghinner** [comment]
@angelik0828  https://deploy-preview-99--did-app.netlify.app/

**Comment by Eghinner** [comment]
It seems that the input is not in the correct position.

https://user-images.githubusercontent.com/71513679/191569847-8ff71d0b-6558-44f0-a3a1-b73c5d5a80b3.mp4

chrome_UvJ6kY5gNy

**Review by Eghinner — APPROVED** [review]
